libs/ktem/ktem/reasoning/simple.py

```diff
@@ -624,7 +648,8 @@ async def ainvoke(  # type: ignore
         self, message: str, conv_id: str, history: list, **kwargs  # type: ignore
     ) -> Document:  # type: ignore
         if self.use_rewrite:
-            rewrite = await self.rewrite_pipeline(question=message)
+            rewrite_pipeline = random.choice(self.rewrite_pipelines)
+            rewrite = await rewrite_pipeline(question=message)
```

Comment:
Why do we want to pick randomly here ? Working with LLMs introduces plenty of non-determinism already, please avoid adding unnecessary randomness.

If you really need it, please make it configurable using an attribute such as `self.use_random_rewrite_pipeline` and also use a method e.g. `self.rewrite_message` for handling choosing rewriting pipeline and produce the rewritten text..